src/Bathymetry.jl

```diff
@@ -244,97 +252,81 @@ function interpolate_bathymetry_in_passes(native_z, target_grid;
-and removing all but the specified number of largest connected regions (which represent the ocean and 
-other possibly disconnected regions like the Mediterranean and the Bering sea).
+Remove independent basins from the bathymetry data stored in `z_data` by identifying connected regions
+below sea level. Basins are removed from smallest to largest until only `keep_major_basins` remain.
 
-# Arguments
-============
+Arguments
+=========
 
 - `z_data`: A 2D array representing the bathymetry data.
-- `connected_regions_allowed`: The maximum number of connected regions to keep. 
-                               Default is `Inf`, which means all connected regions are kept.
+- `keep_major_basins`: The maximum number of connected regions to keep. 
+                       Default is `Inf`, which means all connected regions are kept.
 
 """
-function remove_lakes!(z_data::Field; kw...)
-    data = Array(interior(z_data, :, :, 1))
-    data = remove_lakes!(data; kw...)
+function remove_minor_basins!(Z::Field, keep_major_basins)
+    Zi = interior(Z, :, :, 1)
+    Zi_cpu = on_architecture(CPU(), Zi)
+    remove_minor_basins!(Zi_cpu, keep_major_basins)
+    set!(Z, Z_cpu)
 
-    set!(z_data, data)
-
-    return z_data
+    return Z
 end
 
-function remove_lakes!(z_data; connected_regions_allowed = Inf)
+function remove_minor_basins!(Z, keep_major_basins)
 
-    if connected_regions_allowed == Inf
-        @info "we are not removing lakes"
-        return z_data
+    if !isfinite(keep_major_basins)
+        throw(ArgumentError("`keep_major_basins` must be a finite number!"))
     end
 
-    bathtmp = deepcopy(z_data)
-    batneg  = zeros(Bool, size(bathtmp)...)
-    
-    batneg[bathtmp.<0] .= true
+    water = Z .< 0
     
-    connectivity = ImageMorphology.strel(batneg)
+    connectivity = ImageMorphology.strel(water)
     labels = ImageMorphology.label_components(connectivity)
     
     total_elements = zeros(maximum(labels))
     label_elements = zeros(maximum(labels))
 
-    for i in 1:lastindex(total_elements)
-        total_elements[i] = length(labels[labels .== i])
-        label_elements[i] = i
+    for e in 1:lastindex(total_elements)
+        total_elements[e] = length(labels[labels .== e])
+        label_elements[e] = e
     end
         
-    all_idx = []
-    ocean_idx = findfirst(x -> x == maximum(total_elements), total_elements)
-    push!(all_idx, label_elements[ocean_idx])
-    total_elements = filter((x) -> x != total_elements[ocean_idx], total_elements)
-    label_elements = filter((x) -> x != label_elements[ocean_idx], label_elements)
+    Ib = [] # major basins indexes
+    Iw = findfirst(x -> x == maximum(total_elements), total_elements)
+    push!(Ib, label_elements[Iw])
+    total_elements = filter(x -> x != total_elements[Iw], total_elements)
+    label_elements = filter(x -> x != label_elements[Iw], label_elements)
 
-    for _ in 1:connected_regions_allowed
+    for _ in 1:keep_major_basins
```

Comment:
do we want to have `keep_major_basins` include the largest basin by default or not? That is the difference between having a global ocean using
```
keep_major_basins = 1
```
or 
```
keep_major_basins = 0
```

Reply:
I think `1` makes more sense, otherwise its hard to explain the meaning of the variable. And we should check that its not 0 probably.